docs/extending/code-overview/pattern-matching.rst

```diff
@@ -8,9 +8,9 @@
 Pattern Matching in Evaluation
 ==============================
 
-Pattern matching is used in evaluation of Mathics3 Expression and therefore in Function resolution. Because this process is both time consuming and involved, it is important to understand how it works. We describe some of this process here.
+Pattern matching is used in evaluation of Mathics3 Expression and therefore in Function resolution. Because this process is both time-consuming and involved, it is important to understand how it works. We describe some of this process here.
 
-Based on the ``Head`` of a Mathics3 Expression, the rules stored the ``Definitions`` object for ``Head`` are examined for matches.
+In the evaluation loop, the rules stored in the ``Definition`` object associated with each ``Element`` of an ``Expression`` are examined for matching with the Element (`Downvalues`) or the subexpression (`Upvalues`, `Subvalues`) containing the ``Element``.
```

Comment:
Is this the same? 

In the evaluation loop, each ``Element`` of an ``Expression``  is looked up in the evaluation's ``definitions`` object for the evaluation. Each rule matching the Element (`Downvalues`) or the subexpression (`Upvalues`, `Subvalues`) containing the ``Element`` is considered.

Reply:
Yes, and it is clearer, thanks!